STYLE.md

```diff
@@ -0,0 +1,228 @@
+              }
+
+              if (NULL != foo)
+              {
+                free(foo);
+              }
+              if (NULL != bar)
+              {
+                free(bar);
+              }
+              if (NULL != baz)
+              {
+                free(baz);
+              }
+              return result;
+            }
+
+    - Good:
+
+            void someFunction(int x)
+            {
+              int result = 0;
+              char* foo = NULL;
+              char* bar = NULL;
+              char* baz = NULL;
+
+              if (7 == x)
+              {
+                return EINVAL;
+              }
+
+              foo = func();
+              if (NULL == foo)
+              {
+                result = ENOMEM;
+                goto cleanup;
+              }
+
+              bar = func();
+              if (NULL == bar)
+              {
+                result = ENOMEM;
+                goto cleanup;
+              }
+
+              baz = func();
+              if (NULL == baz)
+              {
+                result = ENOMEM;
+                goto cleanup;
+              }
+
+              result = doSomething(foo, bar, baz);
+
+            cleanup:
+              if (NULL != foo)
+              {
+                free(foo);
+              }
+              if (NULL != bar)
+              {
+                free(bar);
+              }
+              if (NULL != baz)
+              {
+                free(baz);
+              }
+              return result;
+            }
+
+
+# Naming Conventions
+- PascalCase for files, functions, types, and structs
+- camelCase for parameters and variables
+- g_ prefix for global variables
+- UPPER_CASE for macros
+
+# Variables
+-  Variables must always be initialized
+-  For C, variables should be defined at the beginning of the code block they're used:
+    - Bad:
+
+            int foo(int bar)
+            {
+              int x=0; // variable defined at the beginning of a larger block, but it's used only in the smaller block
+              if (1 == bar)
+              {
+                x = 7;
+                return x;
+              }
+              int y = 5; // variable defined not at the beginning of the code block
+              return y;
+            }
+
+    - Good:
+
+            int foo(int bar)
+            {
+              int y = 5;
+              if (1 == bar)
+              {
+                int x=0;
+                x = 7;
+                return x;
+              }
+              return y;
+            }
+
+- Variables must not be reused for different purposes, except for simple index variables in for loops and a function-wide result variable
+- Variable names must be verbose enough to explain their purpose if not obvious
+- Use boolean true and false rather than 1 and 0 for boolean operations
+
+# Error Handling
+- For C, return integer status codes for functions that can fail, typically using standard errno values, and 0 for success
```

Comment:
>For C, return integer status codes for functions that can fail, typically using standard errno values, and 0 for success [](http://example.com/codeflow?start=2&length=120)

... or boolean, for functions that make sense (fictive example: bool IsSomethingEnabled)